frontend/src/utilities/documents.ts

```diff
@@ -16,22 +16,34 @@ import { panicProxy } from "@/utilities/panic-proxy";
 
 const wasm = import("@/../wasm/pkg").then(panicProxy);
 
+class DocumentState {
+	readonly displayName: string;
+
+	constructor(readonly name: string, readonly isSaved: boolean) {
+		this.displayName = `${name}${isSaved ? "" : "*"}`;
+	}
+}
+
 const state = reactive({
-	documents: [] as string[],
+	documents: [] as DocumentState[],
 	activeDocumentIndex: 0,
-	get activeDocument() {
-		return this.documents[this.activeDocumentIndex];
-	},
 });
 
 export async function selectDocument(tabIndex: number) {
 	(await wasm).select_document(tabIndex);
 }
 
 export async function closeDocumentWithConfirmation(tabIndex: number) {
-	selectDocument(tabIndex);
+	const targetDocument = state.documents[tabIndex];
+	if (targetDocument.isSaved) {
+		(await wasm).close_document(tabIndex);
+		return;
+	}
```

Comment:
When will this occur? As I recall the situation here, this response/message `closeDocumentWithConfirmation` should only be sent off from the backend in the case where a tab is being closed which was confirmed by the backend to be unsaved. This doesn't hurt, but it seems unnecessary. However my memory of exactly when this occurs could be spotty... does this code actually change behavior at all?

Reply:
closeDocumentWithConfirmation is called by the Vue frontend. The if statement stops the popup from being displayed if the document is already saved

Reply:
Oh, gotcha. Yes, that makes sense. I had gotten that confused with the other messages (`DisplayConfirmationToCloseDocument` and `DisplayConfirmationToCloseAllDocuments`) which is sent from the backend. This is a good chance 👍. Ready to merge!